configure.ac

```diff
@@ -105,6 +107,16 @@ AC_CHECK_LIB(dl, dlopen, [],
 ])
 fi
 
+dnl handle picky option
+AC_ARG_ENABLE([picky],
+    [AC_HELP_STRING([--enable-picky],
+                    [Enable developer-level compiler pickyness when building @<:@default=no@:>@])])
+AS_IF([test x"$enable_picky" = x"yes" -a x"$GCC" = x"yes"],
```

Comment:
I'm sorry I missed this the first time -- we don't use `-a` because some `test` flavors don't support it.  Instead, we use `&& test` (i.e., a 2nd `test` statement).